monkey/monkey_island/cc/services/config.py

```diff
@@ -567,3 +570,33 @@ def _format_targets_from_flat_config(config: Dict):
         config.pop(flat_subnet_scan_list_field, None)
 
         return formatted_scan_targets_config
+
+    @staticmethod
+    def _format_exploiters_from_flat_config(config: Dict):
+        flat_config_exploiter_classes_field = "exploiter_classes"
+        brute_force_category = "brute_force"
+        vulnerability_category = "vulnerability"
+        brute_force_exploiters = {
+            "MSSQLExploiter",
+            "PowerShellExploiter",
+            "SSHExploiter",
+            "SmbExploiter",
+            "WmiExploiter",
+        }
+
+        formatted_exploiters_config = {"brute_force": [], "vulnerability": []}
+
+        for exploiter in sorted(config[flat_config_exploiter_classes_field]):
+            category = (
+                brute_force_category
+                if exploiter in brute_force_exploiters
+                else vulnerability_category
+            )
+
+            formatted_exploiters_config[category].append(
+                {"name": exploiter, "propagator": (exploiter != "ZerologonExploiter")}
```

Comment:
Similar to the above. Why not use `RUNS_AGENT_ON_SUCCESS`?

Reply:
For many of the same reasons as https://github.com/guardicore/monkey/pull/1657#discussion_r769558408. See also #1605, which proposes a better solution for this.